tests/StacktraceTest.php

```diff
@@ -457,6 +457,19 @@ public function testFromBacktraceWithAnonymousClass(): void
         );
     }
 
+    public function testFromBacktraceWithAnonymousClassTrimsPath(): void
+    {
+        $frame = [
+            'class' => 'class@anonymous' . "\x00" . $this->options->getPrefixes()[0] . '/path/to/file.php:1',
+            'function' => 'callback',
+            'type' => '->',
+        ];
+
+        $frames = Stacktrace::createFromBacktrace($this->options, $this->serializer, $this->representationSerializer, [$frame], 'a.php', 1)->getFrames();
+
+        self::assertSame('class@anonymous' . "\x00" . '/path/to/file.php:1::callback', $frames[1]->getFunctionName());
```

Comment:
Instead of concatenating the three strings, it's enough to wrap the entire string with double quotes